crates/bin/pcli/src/transaction_view_ext.rs

```diff
@@ -0,0 +1,349 @@
+                    '5' => "⠑", // "\u{2585}",
+                    '6' => "⠋", // "\u{2586}",
+                    '7' => "⠛", // "\u{2587}",
+                    '8' => "⠓", // "\u{2588}",
+                    '9' => "⠊", // "\u{2589}",
+                    'A' => "⠊", // "\u{259A}",
+                    'B' => "⠋", // "\u{259B}",
+                    'C' => "⠌", // "\u{259C}",
+                    'D' => "⠍", // "\u{259D}",
+                    'E' => "⠎", // "\u{259E}",
+                    'F' => "⠏", // "\u{259F}",
+                    _ => "",
+                }
+                .to_string()
+            })
+            .collect();
+
+        //format!("\u{001b}[{};{}m{}", fg_color_code, bg_color_code, block_chars)
+        format!("{}", opaque_chars)
+
+    }
+}
+
+// feels like these functions should be extension traits of their respective structs
+// propose moving this to core/keys/src/address/view.rs
+fn format_address_view(address_view: &AddressView) -> String {
+    match address_view {
+        AddressView::Visible { address: _, index, wallet_id: _ } => {
+            format!("[address {:?}]", index.account)
+        },
+         AddressView::Opaque { address } => {
+            // slicing off the first 8 chars to match the plaintext length for aesthetics
+            format!("{}", format_opaque_bytes(&address.to_vec()[..8]))
+        },
+    }
+}
+
+// feels like these functions should be extension traits of their respective structs
+// propose moving this to core/asset/src/value.rs
+fn format_value_view(value_view: &ValueView) -> String {
+    match value_view {
+        ValueView::KnownDenom { amount, denom } => {
+            // TODO: This can be further tweaked depending on what DenomMetadata units should be shown. Leaving as base units for now.
+            format!("{} {}", amount, denom)
+        },
+        ValueView::UnknownDenom { amount, asset_id } => {
+            format!("{} {}", amount, format_opaque_bytes(&asset_id.to_bytes()))
+        },
+    }
+}
+
+
+fn format_fee_view(fee: &Fee) -> String {
+    format!("{} {}", fee.amount(), fee.asset_id())
+}
+
+
+// when handling ValueViews inside of a Visible variant of an ActionView, handling both cases might be needlessly verbose
+// potentially this makes sense as a method on the ValueView enum
+// propose moving this to core/asset/src/value.rs
+fn value_view_amount(value_view: &ValueView) -> Amount {
+    match value_view {
+        ValueView::KnownDenom { amount, .. } | ValueView::UnknownDenom { amount, .. } => {
+            *amount
+        }
+        _ => panic!("Unexpected ValueView variant!"),
+    }
+}
+
+pub trait TransactionViewExt {
+    /// Render this transaction view on stdout.
+    fn render_terminal(&self);
+}
+
+impl TransactionViewExt for TransactionView {
+    fn render_terminal(&self) {
+        println!("Tx Hash"); // Not available here?
+        println!("Tx Sig"); // Probably not needed
+        println!("Anchor"); // Probably not needed
```

Comment:
Metadata placeholders that may not actually be needed

Reply:
I think the tx hash is available, but in a slightly baroque way: clone the `TransactionView`, then "demote" the `TransactionView` into a `Transaction`, then hash that. Not sure it's important to render here, though, since so far this is only used in a command that takes the hash as input.

Reply:
Agree